### ISSUE TITLE
Credit risk classification with deep learning

### ISSUE THREAD

**Issue by dsaada** [rstudio/tensorflow-blog#23]
Hi all,

I would like to puplish this article on R blog TensorFlow,
the link of my Github:

https://dsaada.github.io/CreditRiskTensorFlow/

Thanks,
david.saada@hotmail.fr

David

**Comment by jjallaire** [comment]
We probably need a bit more narrative to publish this. I also can't opine on the approach/results but @skeydan likely can. @skeydan What do you think needs to be added here to make it publish worthy?

**Comment by dsaada** [comment]
#response_container_BBPPID{font-family: initial; font-size:initial; color: initial;}  Hi,Ok i will be more narrative.I wait other comment before to commit.ThxDavid                                                                                                                                                                        Envoyé de mon BlackBerry - l'appareil mobile le plus sécurisé                                                                                                                                                                           De: notifications@github.comEnvoyé: 18 décembre 2018 14:08À: tensorflow-blog@noreply.github.comRépondre à: reply@reply.github.comCc: david.saada27@gmail.com; author@noreply.github.comObjet: Re: [rstudio/tensorflow-blog] Credit risk classification with deep learning (#23)  We probably need a bit more narrative to publish this. I also can't opine on the approach/results but @skeydan likely can. @skeydan What do you think needs to be added here to make it publish worthy?

**Comment by skeydan** [comment]
Hi,

thanks for your interest to publish on the blog. 
Like @jjallaire already said, I would expect more narrative throughout. In particular, I think the following questions should be addressed:

1) (Not a question). This is not a convolutional neural network (like you say in the abstract).

2) What is the motivation of the post? In principle, when showing an example of a straightforward, default-ish feedforward network (no hyperparameter tuning, etc.) I see 2 possible motivations:

- compare performance to "classical" methods for the task at hand. For example, random forests, logistic regression... Then the outcome would be "deep learning does it better", or "does it as well as", etc.

- Use the example to explain to beginners how to use Keras. If that's the motivation, you'd want to explain how you define a model, compile it, train it, etc. Probably this approach would make most sense here.

3) Here I have a list of unordered questions / remarks, whose importance (resp., the way to address them) would vary depending on what's the underlying motivation (see above). 

- How did you arrive at the dropout rate chosen?
- Why so few units for the dense layers? (these are uncommon numbers)
- There is no further improvement after epoch 2, why? 
- You display lots of metrics/indicators, - what do you conclude from them?
- How about the class imbalance, do you see it as a problem?


4) More narrative overall - what kind of dataset is it, where do you download it, why do you preprocess it the way you do?

To sum up: Quoting your last sentence

> Conclusion: this example show how to use Keras API for binary classification.

this brings me back to the question of motivation. Who do you address here? If you address people who want to learn "how to use Keras API for binary classification.", they will need a lot more information.
Such as: What is an activation, what is a loss, etc. 
If you turn it into a "getting started" - type post, you don't have to worry about things like class imbalance etc. ...It all comes down to whom you're addressing.

**Comment by dsaada** [comment]
#response_container_BBPPID{font-family: initial; font-size:initial; color: initial;} Many thanks for your mail,You want asnwer by mail? Or i do the modification and i show you?                                                                                                                                                                        Envoyé de mon BlackBerry - l'appareil mobile le plus sécurisé                                                                                                                                                                           De: notifications@github.comEnvoyé: 18 décembre 2018 14:55À: tensorflow-blog@noreply.github.comRépondre à: reply@reply.github.comCc: david.saada27@gmail.com; author@noreply.github.comObjet: Re: [rstudio/tensorflow-blog] Credit risk classification with deep learning (#23)  Hi,
thanks for your interest to publish on the blog.
Like @jjallaire already said, I would expect more narrative throughout. In particular, I think the following questions should be addressed:

(Not a question). This is not a convolutional neural network (like you say in the abstract).

What is the motivation of the post? In principle, when showing an example of a straightforward, default-ish feedforward network (no hyperparameter tuning, etc.) I see 2 possible motivations:


compare performance to "classical" methods for the task at hand. For example, random forests, logistic regression... Then the outcome would be "deep learning does it better", or "does it as well as", etc.

Use the example to explain to beginners how to use Keras. If that's the motivation, you'd want to explain how you define a model, compile it, train it, etc. Probably this approach would make most sense here.

Here I have a list of unordered questions / remarks, whose importance (resp., the way to address them) would vary depending on what's the underlying motivation (see above).
How did you arrive at the dropout rate chosen?Why so few units for the dense layers? (these are uncommon numbers)There is no further improvement after epoch 2, why?You display lots of metrics/indicators, - what do you conclude from them?How about the class imbalance, do you see it as a problem?
More narrative overall - what kind of dataset is it, where do you download it, why do you preprocess it the way you do?
To sum up: Quoting your last sentence

Conclusion: this example show how to use Keras API for binary classification.

this brings me back to the question of motivation. Who do you address here? If you address people who want to learn "how to use Keras API for binary classification.", they will need a lot more information.
Such as: What is an activation, what is a loss, etc.
If you turn it into a "getting started" - type post, you don't have to worry about things like class imbalance etc. ...It all comes down to whom you're addressing.

**Comment by skeydan** [comment]
>> Or i do the modification and i show you? 

Yeah that's what I meant :-)

**Comment by dsaada** [comment]
#response_container_BBPPID{font-family: initial; font-size:initial; color: initial;}  Ok i will do.ThxDavid                                                                                                                                                                        Envoyé de mon BlackBerry - l'appareil mobile le plus sécurisé                                                                                                                                                                           De: notifications@github.comEnvoyé: 18 décembre 2018 15:03À: tensorflow-blog@noreply.github.comRépondre à: reply@reply.github.comCc: david.saada27@gmail.com; author@noreply.github.comObjet: Re: [rstudio/tensorflow-blog] Credit risk classification with deep learning (#23)  

Or i do the modification and i show you?


Yeah that's what I meant :-)

**Comment by dsaada** [comment]
#response_container_BBPPID{font-family: initial; font-size:initial; color: initial;}  Hi,Could you pls have a look?ThxDavid                                                                                                                                                                        Envoyé de mon BlackBerry - l'appareil mobile le plus sécurisé                                                                                                                                                                           De: david.saada27@gmail.comEnvoyé: 18 décembre 2018 14:10À: reply@reply.github.com; tensorflow-blog@noreply.github.comCc: author@noreply.github.comObjet: Re: [rstudio/tensorflow-blog] Credit risk classification with deep learning (#23)  #ssc26014 #response_container_BBPPID{font-family: initial;font-size: initial;color: initial;}  Hi,Ok i will be more narrative.I wait other comment before to commit.ThxDavid                                                                                                                                                                        Envoyé de mon BlackBerry - l'appareil mobile le plus sécurisé                                                                                                                                                                           De:  notifications@github.comEnvoyé: 18 décembre 2018 14:08À:  tensorflow-blog@noreply.github.comRépondre à:  reply@reply.github.comCc:  david.saada27@gmail.com; author@noreply.github.comObjet: Re: [rstudio/tensorflow-blog] Credit risk classification with deep learning (#23)  We probably need a bit more narrative to publish this. I also can't opine on the approach/results but @skeydan likely can. @skeydan What do you think needs to be added here to make it publish worthy?

**Comment by skeydan** [comment]
Hi,

first of all, thanks again for your interest in publishing on the blog. We've discussed this and we feel that although the exposition is interesting, the text does not fit in completely with other contributed posts.

Asking for contributions, what we mostly have in mind is applications by practitioners, in their respective work areas.  These would normally entail quite a bit of domain specific context and domain-specific "tips and tricks" to get deep learning to work well.
Contributions could also be more technology-related, like exploring lesser used features, or tuning-oriented, striving for optimal performance on a dataset.
When I said this would probably rather make for an introduction for beginners, what I had in mind was an article intended as an introduction to basic feedforward networks, which would have to be quite a bit more detailed, and specifically tailored to presenting core ideas of deep learning (resp. DL frameworks).

I'm sorry I didn't reply like that in the beginning and caused you to spend more time. On the other hand, you could certainly publish it yourself, e.g., on github pages. Like that, it won't have been wasted time!

Best,
Sigrid

**Comment by dsaada** [comment]
hi,

Thanks for your email. So what can I do for publish in your blog? follow
your instruction?
thanks
David


Le jeu. 20 déc. 2018 à 15:38, Sigrid Keydana <notifications@github.com> a
écrit :

**Comment by skeydan** [comment]
Sorry, what I meant was we decided otherwise (for the above reasons). 
But you can always publish it yourself - like that, it's publicly visible too.
And again, sorry for not having responded that way from the outset.
